src/media_tools/av_parsers.c

```diff
@@ -1545,6 +1545,595 @@ GF_Err gf_m4a_write_config(GF_M4ADecSpecInfo *cfg, char **dsi, u32 *dsi_size)
+	if (dw != GF_4CC('A', 'V', '0', '1')) {
+		GF_LOG(GF_LOG_ERROR, GF_LOG_CODING, ("[IVF] Wrong codec FourCC. Only 'AV01' supported, got '%c%c%c%c'\n", FourCC[3], FourCC[2], FourCC[1], FourCC[0]));
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+	dw = gf_bs_read_u16_le(bs);
+	state->width = state->width < dw ? dw : state->width;
+	dw = gf_bs_read_u16_le(bs);
+	state->height = state->height < dw ? dw : state->height;
+	dw = gf_bs_read_u32_le(bs); //time_base.numerator
+	dw = gf_bs_read_u32_le(bs); //time_base.denominator
+
+	gf_bs_read_u64(bs); //skip
+
+	return GF_OK;
+}
+
+GF_Err gf_media_aom_parse_ivf_frame_header(GF_BitStream *bs, u64 *frame_size)
+{
+	assert(frame_size);
+	*frame_size = gf_bs_read_u32_le(bs);
+	if (*frame_size > 256 * 1024 * 1024) {
+		GF_LOG(GF_LOG_ERROR, GF_LOG_CODING, ("[IVF] Wrong frame size %u\n", *frame_size));
+		*frame_size = 0;
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+
+	gf_bs_read_u64(bs); //skip
+
+	return GF_OK;
+}
+
+static GF_Err av1_parse_obu_header(GF_BitStream *bs, ObuType *obu_type, Bool *obu_extension_flag, Bool *obu_has_size_field, u8 *temporal_id, u8 *spatial_id)
+{
+	Bool forbidden = gf_bs_read_int(bs, 1);
+	if (forbidden) {
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+	assert(obu_type && obu_extension_flag && obu_has_size_field && temporal_id && spatial_id);
+	*obu_type = gf_bs_read_int(bs, 4);
+	*obu_extension_flag = gf_bs_read_int(bs, 1);
+	*obu_has_size_field = gf_bs_read_int(bs, 1);
+	gf_bs_read_int(bs, 1); /*obu_reserved_1bit*/
+	if (*obu_extension_flag) {
+		*temporal_id = gf_bs_read_int(bs, 3);
+		*spatial_id = gf_bs_read_int(bs, 2);
+		/*extension_header_reserved_3bits = */gf_bs_read_int(bs, 3);
+	}
+
+	return GF_OK;
+}
+
+const char *av1_get_obu_name(ObuType obu_type)
+{
+	switch (obu_type) {
+	case OBU_SKIP: return "skip";
+	case OBU_SEQUENCE_HEADER: return "seq_header";
+	case OBU_TEMPORAL_DELIMITER: return "delimiter";
+	case OBU_FRAME_HEADER: return "frame_header";
+	case OBU_TILE_GROUP: return "tile_group";
+	case OBU_METADATA: return "metadata";
+	case OBU_FRAME: return "frame";
+	case OBU_REDUNDANT_FRAME_HEADER: return "redundant_frame_header";
+	case OBU_PADDING: return "padding";
+	default: return "unknown";
+	}
+}
+
+Bool av1_is_obu_header(ObuType obu_type) {
+	switch (obu_type) {
+	case OBU_SEQUENCE_HEADER:
+	case OBU_METADATA:
+		return GF_TRUE;
+	default:
+		return GF_FALSE;
+	}
+}
+
+static Bool av1_is_obu_frame(ObuType obu_type) {
+	switch (obu_type) {
+	case OBU_TEMPORAL_DELIMITER:
+	case OBU_PADDING:
+	case OBU_REDUNDANT_FRAME_HEADER:
+		return GF_FALSE;
+	default:
+		return GF_TRUE;
+	}
+}
+
+u64 leb128(GF_BitStream *bs, u8 *opt_Leb128Bytes) {
+	u64 value = 0;
+	u8 Leb128Bytes = 0, i = 0;
+	for (i = 0; i < 8; i++) {
+		u8 leb128_byte = gf_bs_read_u8(bs);
+		value |= ((leb128_byte & 0x7f) << (i * 7));
+		Leb128Bytes += 1;
+		if (!(leb128_byte & 0x80)) {
+			break;
+		}
+	}
+
+	if (opt_Leb128Bytes) {
+		*opt_Leb128Bytes = Leb128Bytes;
+	}
+	return value;
+}
+
+#define OBU_COPY \
```

Comment:
Probably worth having a check on obu_length, no?

Reply:
Which kind of check would you imagine?